src/ast.rs

```diff
@@ -1424,6 +1465,12 @@
         Some(self)
     }
 }
+impl CheckParse for ElseifClause {
+    fn can_be_parsed(pop: &mut Populator<'_>) -> bool {
```

Comment:
I think you should squash all commits into one, to simplify things overall.
I don't want to read several different approaches in full, only the best one.